app/scripts/controllers/network/createJsonRpcClient.js

```diff
@@ -32,3 +47,16 @@ function createChainIdMiddleware (chainId) {
     return next()
   }
 }
+
+/**
+ * For use in tests only.
+ * Adds a delay to `eth_estimateGas` calls.
+ */
+function createEstimateGasDelayMiddleware () {
+  return createAsyncMiddleware(async (req, _, next) => {
+    if (req.method === 'eth_estimateGas') {
+      await new Promise((resolve) => setTimeout(resolve, 2000))
```

Comment:
This `resolve` call doesn't need to return a value?

Reply:
Nah, as written it's just a `Promise<undefined>`, and we of course don't do anything with the return value anyway.